bootstrap/internal/cloudinit/cloudinit.go

```diff
@@ -121,3 +124,37 @@ func generate(kind string, tpl string, data interface{}) ([]byte, error) {
 
 	return out.Bytes(), nil
 }
+
+func cleanupAdditionalCloudInit(cloudInitData string) (string, error) {
+	m := make(map[string]interface{})
+
+	err := yaml.Unmarshal([]byte(cloudInitData), m)
+	if err != nil {
+		return "", err
+	}
+
+	// Remove the runcmd section
+	delete(m, "runcmd")
```

Comment:
Instead of doing delete like this could we store the fields we want to remove in a slide and then iterate over it?

Reply:
Done!